src/service/cid-api.js

```diff
@@ -85,7 +86,8 @@ export class GoogleCidApi {
       if (token === TokenStatus.OPT_OUT) {
         return null;
       }
-      if (token === TokenStatus.ERROR) {
+      // Token is in a special state, fallback to existing cookie
+      if (token && token[0] === '$') {
```

Comment:
I think the idea here is to do a re-request immediately when `referer == *.cdn.ampproject.org && token == '$NOT_FOUND'`. For all the other cases, it is okay to get cookie value.